src/vue/modules/document-upload-form/index.vue

```diff
@@ -0,0 +1,200 @@
+          />
+        </div>
+      </div>
+
+      <div class="app__form-actions document-upload-form__actions">
+        <form-confirmation
+          v-if="formMixin.isConfirmationShown"
+          :is-pending="isDocumentUploading"
+          @ok="submit"
+          @cancel="hideConfirmation"
+        />
+        <button
+          v-else
+          v-ripple
+          type="submit"
+          class="app__button-raised"
+          :disabled="formMixin.isDisabled"
+        >
+          {{ 'document-upload-form.upload-btn' | globalize }}
+        </button>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script>
+import FormMixin from '@/vue/mixins/form.mixin'
+
+import { Wallet, base, errors } from '@tokend/js-sdk'
+import { DOCUMENT_TYPES } from '@/js/const/document-types.const'
+
+import { documentContainer, required } from '@validators'
+
+import { initApi } from './_api'
+import { initConfig } from './_config'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { FileUtil } from '@/js/utils/file.util'
+import { CryptoUtil } from './utils/crypto.util'
+
+import { DocumentUploader } from './helpers/document-uploader'
+
+import { mapActions, mapMutations } from 'vuex'
+import { types } from './store/types'
+
+const EVENTS = {
+  submit: 'submit',
+}
+
+export default {
+  name: 'document-upload-form-module',
+  mixins: [FormMixin],
+  props: {
+    /**
+     * @property config - the config for component to use
+     * @property config.horizonURL - the url of horizon server (without version)
+     * @property config.storageURL - the url of file storage server
+     */
+    config: {
+      type: Object,
+      required: true,
+    },
+    wallet: {
+      type: Wallet,
+      required: true,
+    },
+  },
+
+  data: _ => ({
+    form: {
+      document: null,
+      description: '',
+    },
+    isDocumentUploading: false,
+    DOCUMENT_TYPES,
+  }),
+  validations: {
+    form: {
+      document: { documentContainer },
+      description: { required },
+    },
+  },
+
+  created () {
+    initApi(this.wallet, this.config)
+    initConfig(this.config)
+    this.setAccountId(this.wallet.accountId)
+  },
+
+  methods: {
+    ...mapMutations('upload-form', {
+      setAccountId: types.SET_ACCOUNT_ID,
+      setDocumentAccountId: types.SET_DOCUMENT_ACCOUNT_ID,
+    }),
+
+    ...mapActions('upload-form', {
+      createAccount: types.CREATE_ACCOUNT,
+      createChangeRoleRequest: types.CREATE_CHANGE_ROLE_REQUEST,
+      createBlob: types.CREATE_BLOB,
+    }),
+
+    async submit () {
+      if (!this.isFormValid()) return
+
+      this.isDocumentUploading = true
+      this.disableForm()
+
+      try {
+        const documentText = await FileUtil.getText(this.form.document.file)
+        const documentHash = await CryptoUtil.hashMessage(documentText)
+
+        const accountId = await this.deriveAccountId(documentText)
```

Comment:
`getAttachedAccountId` or `getAccountIdFromDocHash`

Reply:
Renamed to `getAccountIdFromDocHash`